common/lib/capa/capa/capa_problem.py

```diff
@@ -191,6 +193,48 @@ def __init__(self, problem_text, id, capa_system, state=None, seed=None):
 
         self.extracted_tree = self._extract_html(self.tree)
 
+    def make_xml_compatible(self, tree):
+        """
+        Adjust tree xml in-place for compatibility, such as
+        supporting an old xml format by translating it to a new format.
```

Comment:
Actually... this function only changes the format for two types of inputs.

I would move the block comments you have inline with the code and move them to the docstring, along with an explanation about "this is compatibility for the new <additional answer> type" (something better than that, obvs)